### ISSUE TITLE
MaxClientsPerUser not enforced for SFTP logins when mod_digest enabled

### ISSUE THREAD

**Issue by tomfite** [proftpd/proftpd#750]
### What I Did

After including `mod_digest` in my build, `mod_auth` is not enforcing any of the connection limits (MaxClientsPerHost, MaxClientsPerUser). I've added debug lines to the `auth_count_scoreboard` method in `mod_auth.c` and these do not get hit at all when mod_digest is compiled and a login via SFTP is made. 

I've confirmed that removing `mod_digest` from the build enforces connection limits for SFTP connections.

### What I Expected/Wanted

Expected behavior is `auth_count_scoreboard` code is hit when `mod_digest` is enabled. Connection limits should be enforced when connecting via SFTP. 

This only affects SFTP connections, connection limits are enforced with the `mod_digest` enabled for FTP connections.

### ProFTPD Version and Configuration

```
 Compile-time Settings:
  Version: 1.3.6 (stable)
  Platform: LINUX [Linux 3.10.0-327.el7.x86_64 x86_64]
  Built: Thu Sep 20 2018 08:44:20 PDT
  Built With:
    configure  '--localstatedir=/var/run' '--with-pkgconfig=libdata/pkgconfig' '--sysconfdir=/usr/local/etc' '--enable-dso' '--disable-sendfile' '--disable-ipv6' '--enable-nls' '--enable-ctrls' '--with-modules=mod_evapi:mod_readme:mod_rewrite:mod_wrap2:mod_wrap2_file:mod_wrap2_sql:mod_site_misc:mod_unique_id:mod_sql:mod_sql_mysql:mod_sql_passwd:mod_tls:mod_ctrls_admin:mod_quotatab:mod_quotatab_file:mod_quotatab_sql:mod_sftp:mod_sftp_sql:mod_sftp_pam:mod_ifsession:mod_statcache' '--with-includes=:/include:/usr/include/curl:/mnt/ev_source/exavault_proftpd-mod-evapi/jansson:/usr/local/include:/usr/local/include/mysql' '--with-libraries=/usr/local/lib:/usr/local/lib/mysql' '--prefix=/usr/local' '--mandir=/usr/local/man' '--infodir=/usr/local/info/' '--disable-xattr' '--enable-tests' 'LIBS=-lcurl'

  CFLAGS: -g2 -O2 -Wall -fno-omit-frame-pointer
  LDFLAGS: -L$(top_srcdir)/lib  -rdynamic -L/usr/local/lib/mysql -L/usr/local/lib -L/usr/lib64
  LIBS:  -lssl -lcrypto -L$(top_srcdir)/lib/libcap -lcap  -lssl  -lm -lmysqlclient  -ljansson -lz  -lpam  -lcrypto -lsupp -lcrypt -ldl -lcurl

  Files:
    Configuration File:
      /usr/local/etc/proftpd.conf
    Pid File:
      /var/run/proftpd.pid
    Scoreboard File:
      /var/run/proftpd.scoreboard
    Header Directory:
      /usr/local/include/proftpd
    Shared Module Directory:
      /usr/local/libexec

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    - Autoshadow support
    + Controls support
    + curses support
    - Developer support
    + DSO support
    - IPv6 support
    + Largefile support
    - Lastlog support
    - Memcache support
    + ncursesw support
    + NLS support
    - Redis support
    - Sodium support
    + OpenSSL support (FIPS enabled)
    - PCRE support
    - POSIX ACL support
    + Shadow file support
    - Sendfile support
    + Trace support
    - xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```

Relevant proftpd config:
```
ServerType standalone
ScoreboardFile /var/run/proftpd.scoreboard

# disable reverse DNS for speed
UseReverseDNS off

# 03 May 2018: Upping this from default value of 5, which causes "connection refused" if more than
# this number connect at the same time
tcpBackLog 50

# SystemLog: redirect what otherwise goes to /var/log/messages. Also see SyslogLevel, below.
SystemLog /var/log/container/proftpd-log.log
SyslogLevel warn

<IfModule mod_statcache.c>
    StatCacheEngine off
</IfModule>

# bind to exact IP's of service
SocketBindTight on

# Set TLS Protocol
TLSProtocol TLSv1.2 TLSv1.1 TLSv1

SocketOptions rcvbuf 1048456 sndbuf 1048456

# MaxInstances
MaxInstances 1500

# turn off the 'server config' default server
Port 0

# Enable the Controls for ftpdctl to function
<IfModule mod_ctrls.c>
        ControlsEngine          on
        ControlsACLs            all allow group wheel
        ControlsMaxClients      5
        ControlsLog             /var/log/container/proftpdctl.log
        ControlsInterval        10
        ControlsSocketACL       allow group wheel
        ControlsSocket          /var/run/proftpd.sock
        ControlsSocketOwner     root wheel

   <IfModule mod_ctrls_admin.c>
        AdminControlsACLs all allow group wheel
   </IfModule>
</IfModule>

<Global>

# Enable proxy support, this allows proftpd to recognize the client IP from haproxy
<IfModule mod_proxy_protocol.c>
  ProxyProtocolEngine on
  ProxyProtocolTimeout 3sec
  ProxyProtocolVersion haproxyV1

  # Necessary to allow data transfers
  AllowForeignAddress on
</IfModule>

# By default, use binary mode for transfers. ASCII can corrupt files by converting carriage-return
# to linefeed characters (unix vs windows)
DefaultTransferMode binary

# And if ASCII mode is requested, don't convert line endings to avoid corrupting files.
TransferOptions IgnoreASCII

# Do not check for .ftpaccess files
AllowOverride off

# Force DeleteAbortedStores off, this fixes an issue causing files that are resumed to get
# deleted on connection timeout
DeleteAbortedStores off

# Turn off the default usage of /var/log/xferlog
TransferLog NONE

# Allows for noncompliant Active Transfers
# when RootRevoke is enabled. (http://bugs.proftpd.org/show_bug.cgi?id=3731)
RootRevoke UseNonCompliantActiveTransfers

# Umask 006 007 gives RWXRWX--- for folders, and RW-RW---- for files
Umask 006 007

CommandBufferSize 512

# Set the user and group under which the server will run.
User nobody
Group nobody

# disable ident lookups for speed. Note we have to put this directive in the <Global> context,
# not the server config content (where UseReverseDNS goes)
IdentLookups off

# Set times to GMT
TimesGMT on

# Turn off WTMP logging
WtmpLog off

# Make sure every user is "jailed" (chrooted) into their home directory
DefaultRoot ~

# Normally, we want files to be overwriteable.
AllowOverwrite on

# Show hidden files, etc.
ListOptions "-a"

# Set Idle Timeouts
TimeoutIdle 3600
TimeoutNoTransfer 3600

# Allow FTP Resume
AllowStoreRestart on
AllowRetrieveRestart on

AllowChrootSymlinks on

# Setting MaxClientsPerHost and MaxClientsPerUser
MaxClientsPerHost               40
MaxClientsPerUser               30

# disallow root logins
RootLogin off

# do not require a shell; these are virtual users
RequireValidShell off

# disable PAM, if installed. It just causes problems, and we don't need it.
<IfModule mod_sftp_pam.c>
   SFTPPAMEngine off
</IfModule>
<IfModule mod_auth_pam.c>
   AuthPAM off
</IfModule>

# Only authenticate via mod_sql
AuthOrder mod_sql.c

[db / site configuration omitted]
...

**Comment by Castaglia** [comment]
I have reproduced this behavior -- but it does not require the `mod_digest` module to be enabled.  I think this behavior might be specific to `mod_sftp`; investigating further.

**Comment by Castaglia** [comment]
I am now able to reproduce this locally; my debug logging of `auth_count_scoreboard()` are *not* seen when `mod_digest` is included.  Curious.  Now to see what might be happening.

**Comment by Castaglia** [comment]
Fixed in `master`, and backported to the 1.3.6 branch.  Thanks!
